t/proxyunits.t

```diff
@@ -228,11 +228,40 @@ sub proxy_test {
     is(scalar <$ps>, "ok\r\n", "got data back");
     is(scalar <$ps>, "END\r\n", "got end string");
 
-    like(<$w>, qr/ts=(\S+) gid=\d+ type=proxy_backend error=trailingdata name=127.0.0.1 port=\d+ depth=0 rbuf=garbage/, "got backend error log line");
+    like(<$w>, qr/ts=(\S+) gid=\d+ type=proxy_backend error=trailingdata name=127.0.0.1 port=\d+ label=b\d+ depth=0 rbuf=garbage/, "got backend error log line");
 
     $mbe[0] = accept_backend($mocksrvs[0]);
 }
 
+{
+    note("Test proxyevents for a backend without a label");
+
+    my $msrv_nolabel = mock_server(11414);
+    ok(defined $msrv_nolabel, "mock server with no label created");
+    my $mbe_nolabel = accept_backend($msrv_nolabel);
+
+    # If END is missing, the backend should respond with an error to the client,
+    # which will emit a proxy event.
+    my $w = $p_srv->new_sock;
+    print $w "watch proxyevents\n";
+    is(<$w>, "OK\r\n", "watcher enabled");
+
+    # write a request to proxy.
+    print $ps "get /nolabel/a\r\n";
+
+    # verify request is received by backend.
+    is(scalar <$mbe_nolabel>, "get /nolabel/a\r\n", "get passthrough");
+
+    # write a response with partial data.
+    print $mbe_nolabel "VALUE /nolabel/a 0 2\r\nhi\r\nEN";
+
+    # verify the error response from proxy
+    is(scalar <$ps>, "SERVER_ERROR backend failure\r\n", "backend failure error");
+
+    # verify a proxy event logline is received with empty label
+    like(<$w>, qr/ts=(\S+) gid=\d+ type=proxy_backend error=timeout name=127.0.0.1 port=11414 label= depth=1 rbuf=EN/, "got backend error log line");
```

Comment:
small bit: when possible we should avoid tripping timeouts when testing error paths, since it makes the tests take longer to run if you do it often.

Maybe some junk return data from the backend instead?

Unless I misread, sorry.